fairseq/trainer.py

```diff
@@ -344,7 +344,7 @@ def maybe_no_sync():
             self._prev_grad_norm = grad_norm
 
             # take an optimization step
-            self.optimizer.step()
+            # self.optimizer.step()  # XLA: this is done by torch_xla
```

Comment:
Maybe adding a comment which explain where precisely this happens?
I see a function removed here, and nowhere close a matching `xm.optimizer_step()` call.